### PR TITLE
Reject improper provider behavior

### PR DESCRIPTION
**Pull request by willscott**
Addresses #193

### CODE REVIEW

**Comment by bemasc** [comment]
:+1:

File: spec/src/provider.spec.js [hunk 1]
```diff
@@ -101,6 +101,93 @@ describe("Provider", function() {
...
+
+    port.onMessage('default', {to: 'testInst', type:'message', message: {
+      'action': 'method',
+      'type': 'm1',
+      'text': ['mystr'],
+      'reqId': 1
+    }});
+  });
+
+  iit("rejects method calls when the provider doesn't return a promise", function (done) {
```
Comment (ryscheng) [inline spec/src/provider.spec.js]: lingering iit?

File: spec/src/provider.spec.js [hunk 2]
```diff
@@ -101,6 +101,93 @@ describe("Provider", function() {
...
+    port.onMessage('default', {
+      channel: 'message'
+    });
+
+    port.onMessage('default', {to: 'testInst', type:'message', message:{
+      'type': 'construct',
+    }});
+
+    port.on('message', function(n) {
+      expect(n.message.error).toContain('hi.myerr');
```
Comment (ryscheng) [inline spec/src/provider.spec.js]: Is there more of an end-to-end way to test this? Does this expect guarantee that the Promise will reject?

File: spec/src/provider.spec.js [hunk 3]
```diff
@@ -101,6 +101,93 @@ describe("Provider", function() {
...
+    }});
+
+    port.onMessage('default', {to: 'testInst', type:'message', message: {
+      'action': 'method',
+      'type': 'm1',
+      'text': ['mystr'],
+      'reqId': 1
+    }});
+
+    expect(called).toEqual(true);
```
Comment (ryscheng) [inline spec/src/provider.spec.js]: This is super confusing. called is never set in this test?

File: spec/src/provider.spec.js [hunk 4]
```diff
@@ -101,6 +101,93 @@ describe("Provider", function() {
...
+
+    port.on('message', function(n) {
+      expect(n.message.error.length).toBeGreaterThan(0);
```
Comment (ryscheng) [inline spec/src/provider.spec.js]: What is the behavior when n.message.error contains more than 1 error? How does the Promise.reject know what to return?

File: spec/src/provider.spec.js (hunk 2 above)
Comment (willscott) [inline spec/src/provider.spec.js]: the fact that ```error``` is set means that the response is rejected.

File: spec/src/provider.spec.js (hunk 4 above)
Comment (willscott) [inline spec/src/provider.spec.js]: n.message.error is a string. if it's defined, then there's an error.

**Comment by willscott** [comment]
@ryscheng - remaining points were valid & addressed in code.

**Comment by ryscheng** [comment]
:+1:

**Pull request merged**